### PR TITLE
Topology napp first draft

### PR DESCRIPTION
**Pull request by renanrodrigo**
First documentation draft for the topology NApp, for reviewing.

### CODE REVIEW

**Comment by beraldoleal** [comment]
@renanrodrigo Tomorrow, @cemsbr and @diraol are finishing the new REST API Documentation generator using swagger. Our napps repository are going to show automatically any rest decorator.

Can you please, change this PR in order to the new format ? So we can see the documentation directly on our napps server?

**Comment by renanrodrigo** [comment]
Sure. Will be done.

File: napps/kytos/of_topology/README.rst [hunk 1]
```diff
@@ -37,239 +29,49 @@ If you are going to install kytos-napps from source code, all napps will be
 installed by default (just remember you need to enable the ones you want
 running).
 
-REST API
-========
+Listened Events
+===============
+
+'kytos/of_core.v0x01.messages.in.ofpt_packet_in'
+------------------------------------------------
```
Comment (beraldoleal) [inline napps/kytos/of_topology/README.rst]: This should be able to handle 1.0 and 1.3 packet_in. `v0x0[14]` for instance.

File: napps/kytos/of_topology/README.rst [hunk 2]
```diff
@@ -37,239 +29,49 @@ If you are going to install kytos-napps from source code, all napps will be
...
+PacketIn events are analyzed to look for end hosts in the topology. As
+**of_lldp** discovers links between switches, **of_topology** ignores LLDP
+packets, making use of any other message coming from a UNI (User to Network
+Interface to update information about the host connected at the other end.
+
+'kytos/of_core.v0x01.messages.in.ofpt_port_status'
+--------------------------------------------------
```
Comment (beraldoleal) [inline napps/kytos/of_topology/README.rst]: This should be able to handle 1.0 and 1.3 packet_in. `v0x0[14]` for instance.

File: napps/kytos/of_topology/README.rst [hunk 3]
```diff
@@ -37,239 +29,49 @@ If you are going to install kytos-napps from source code, all napps will be
...
+NApps of the topology change.
+
+
+Generated Events
+================
 
-Topology
---------
+'kytos/of_topology.change.port_down
+-----------------------------------
```
Comment (beraldoleal) [inline napps/kytos/of_topology/README.rst]: I would suggest `of_topology.change.port.down`

File: napps/kytos/of_topology/README.rst [hunk 4]
```diff
@@ -37,239 +29,49 @@ If you are going to install kytos-napps from source code, all napps will be
...
+                             dpid: <switch.dpid>,
+                             port_description: {<description of the port>}
+                           })
 
-Network Topology Example
-------------------------
 
-A real example of network topology with 3 hosts and 3 switches is
-represented below.
+'kytos/of_topology.change.port_up
```
Comment (beraldoleal) [inline napps/kytos/of_topology/README.rst]: Same here... `of_topology.change.port.up`

File: napps/kytos/of_topology/README.rst [hunk 5]
```diff
@@ -37,239 +29,49 @@ If you are going to install kytos-napps from source code, all napps will be
...
-             "target":"00:00:00:00:00:00:00:01:2"
-          },
-          {
-             "type":"interface",
-             "source":"00:00:00:00:00:00:00:03",
-             "target":"00:00:00:00:00:00:00:03:65534"
-          }
-       ]
-    }
+event = KytosEvent(name='kytos/of_topology.change.port_down',
```
Comment (beraldoleal) [inline napps/kytos/of_topology/README.rst]: Please add this line:

`The event is an instance of KytosEvent, with the following data:`

File: napps/kytos/topology/README.rst [hunk 6]
```diff
@@ -0,0 +1,350 @@
+Overview
+========
+
+This NApp is responsible for keeping and returning the network topology in a
+protocol-independent way.The topology is built based on the datapaths and hosts
+known to the controller.
+Topology is exported in JSON format via REST endpoints.
```
Comment (beraldoleal) [inline napps/kytos/topology/README.rst]: Please add the following lines (something like this):

```if you want to get information about the topology, you should probably use this NApp. This way, you will be able to operate independently of the protocol below.```

File: napps/kytos/topology/README.rst [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+
+All of the Kytos Network Applications are located in the NApps online repository.
+To install this NApp, run:
+
+.. code:: shell
+
+   $ kytos napps install kytos/topology
+
+
+Settings.py paths
```
Comment (beraldoleal) [inline napps/kytos/topology/README.rst]: s/*/Configuring/

File: napps/kytos/topology/README.rst [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+                   "reliability": 0.98
+               }
+            }
+           "1-to-3":{
+               "links": ["1to2","2:3-3:1"],
+               "metrics": {
+                   "custom_cost": 15
+               }
+            }
+          }
```
Comment (beraldoleal) [inline napps/kytos/topology/README.rst]: Ok, I have some suggestions here, but I will discuss with @diraol first.

File: napps/kytos/topology/README.rst [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+
+
+REST API
+========
+
```
Comment (beraldoleal) [inline napps/kytos/topology/README.rst]: Now, all our NApps has the .yml file where the documentation will be displayed on our napps repository.

So all README files with REST API section should point to the NApp repository link.

**Comment by beraldoleal** [comment]
@renanrodrigo @diraol please note that this should not be merged anymore to legacy/of_topoloy. But instead, divided between kytos/of_topology and kytos/topology.

**Comment by diraol** [comment]
I've migrated our current `of_topology` source code to `https://github.com/kytos/of_topology` and also I've made the changes proposed by @renanrodrigo and @macartur, considering @beraldoleal comments over here.

**Comment by beraldoleal** [comment]
@diraol so we can close this, right ?

**Pull request closed without merging**